parquet-column/src/main/java/org/apache/parquet/schema/Type.java

```diff
@@ -283,7 +283,7 @@ protected boolean equals(Type other) {
 
   @Override
   public boolean equals(Object other) {
-    if (!(other instanceof Type) || other == null) {
```

Comment:
Why is this required? The test you wrote fails when I don't apply the change on PrimitiveType, but doesn't fail if I don't apply this.

Reply:
Actually I think null check is not needed here at all: https://docs.oracle.com/javase/specs/jls/se10/html/jls-15.html#jls-15.20.2